Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -568,20 +568,16 @@ void AppendPipeline::resetParserState()
     ASSERT(isMainThread());
     GST_DEBUG_OBJECT(pipeline(), "Handling resetParserState() in AppendPipeline by resetting the pipeline");
 
-    // FIXME: Implement a flush event-based resetParserState() implementation would allow the initialization segment to
-    // survive, in accordance with the spec.
-
     // This function restores the GStreamer pipeline to the same state it was when the AppendPipeline constructor
-    // finished. All previously enqueued data is lost and the demuxer is reset, losing all pads and track data.
+    // finished. All previously enqueued data is lost and the demuxer is flushed, but retains the configuration from the
+    // last received init segment, in accordance to the spec.
 
     // Unlock the streaming thread.
     m_taskQueue.startAborting();
 
-    // Reset the state of all elements in the pipeline.
-    assertedElementSetState(m_pipeline.get(), GST_STATE_READY);
-
-    // Set the pipeline to PLAYING so that it can be used again.
-    assertedElementSetState(m_pipeline.get(), GST_STATE_PLAYING);
+    gst_element_send_event(m_appsrc.get(), gst_event_new_flush_start());
+    gst_element_send_event(m_appsrc.get(), gst_event_new_flush_stop(true));
```

Comment:
A flushing seek already triggers a flush. Why do we need an additional manual flush before?

Reply:
The problem with a flushing seek is that [qtdemux is partially ignoring the seek event because of this code](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/subprojects/gst-plugins-good/gst/isomp4/qtdemux.c#L1653).

<del>However, just sending the flush start/stop events and a [0, infinity) segment event without the seek causes a crash here.</del> Oh, I can't reproduce the crash anymore!

It turns out that with just this flush and segment code I can achieve the same without needing the seek (my original goal):

```
    gst_element_send_event(m_appsrc.get(), gst_event_new_flush_start());
    gst_element_send_event(m_appsrc.get(), gst_event_new_flush_stop(true));

    GstSegment segment;
    gst_segment_init(&segment, GST_FORMAT_BYTES);
    gst_element_send_event(m_appsrc.get(), gst_event_new_segment(&segment));
```